api/python/ai/chronon/cli/tui/widgets.py

```
@@ -0,0 +1,573 @@
+            if step in visited:
+                return
+            visited.add(step)
+            
+            # Check if this step has any unfinished dependencies
+            has_blocking_deps = False
+            
+            for dep in step.step_dependencies:
+                if dep.node_name in self.node_statuses:
+                    for upstream_step in self.node_statuses[dep.node_name]:
+                        if (upstream_step.status != "finished" and 
+                            self._steps_overlap(upstream_step, dep.start_date, dep.end_date)):
```

Flake8 (7.2.0)

[error] visually indented line with same indent as next logical line

(E129)